pkg/serviceaccount/tokens_controller.go

```diff
@@ -240,8 +244,16 @@ func (e *TokensController) secretDeleted(obj interface{}) {
 		return
 	}
 
-	if _, err := e.removeSecretReferenceIfNeeded(serviceAccount, secret.Name); err != nil {
-		glog.Error(err)
+	for i := 1; i <= NumServiceAccountRemoveReferenceRetries; i++ {
```

Comment:
> In that case, is it important to remove non-existing secrets from this list?

In this case, it's not critical, just bothers my referential-integrity-tuned sensibilities. I'm ok tabling this for the moment and revisiting